### PR TITLE
[datadog][v2.1.2] chart creation

### PR DESCRIPTION
**Pull request by clamoriniere**
Add the latest (`v2.1.2`) stable `datadog` chart from "https://github.com/helm/charts"

### CODE REVIEW

**Comment by guangbochen** [comment]
@clamoriniere thanks for the submission, can u please help to split this PR into 2 commits so it's easier to review, also for Rancher's catalogs chart both questions.yaml and app-readme.md is necessary for the UI rendering, they are both exist in the existing version of datadog chart. 

1. 1st commit: Copy the older version to a new folder
2. 2nd commit: Changes made to the older version to update it.

**Comment by clamoriniere** [comment]
Hi @guangbochen 
I updated the PR with the 2 commits. Thank you for reviewing this PR

File: charts/datadog/v2.0.14/Chart.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+version: 2.0.14
+appVersion: "7"
+description: Datadog Agent
 keywords:
 - monitoring
 - alerting
 - metric
 home: https://www.datadoghq.com
-icon: file://../dd_logo_70x75.png
+icon: https://datadog-live.imgix.net/img/dd_logo_70x75.png
```
Comment (guangbochen) [inline charts/datadog/v2.0.14/Chart.yaml]: u may keep using the existing icon `icon: file://../dd_logo_70x75.png` so the air-gapped environment user can view it properly

File: charts/datadog/v2.0.14/requirements.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 dependencies:
   - name: kube-state-metrics
-    version: ~2.2.3
+    version: ~2.3.1
```
Comment (guangbochen) [inline charts/datadog/v2.0.14/requirements.yaml]: you need to run `helm dep update` and push the updated charts
e.g, refer to https://github.com/rancher/charts/tree/master/charts/datadog/v1.33.0/charts

File: charts/datadog/v2.0.14/Chart.yaml (hunk 1 above)
Comment (clamoriniere) [reply charts/datadog/v2.0.14/Chart.yaml]: done

File: charts/datadog/v2.0.14/requirements.yaml (hunk 2 above)
Comment (clamoriniere) [reply charts/datadog/v2.0.14/requirements.yaml]: done

**Comment by clamoriniere** [comment]
hello @guangbochen 
Could you please have another review.
Thanks

**Pull request merged**